Userland/Libraries/LibC/malloc.cpp

```diff
@@ -345,12 +380,45 @@ static void free_impl(void* ptr)
 
 [[gnu::flatten]] void* malloc(size_t size)
 {
-    void* ptr = malloc_impl(size, CallerWillInitializeMemory::No);
+    auto result = malloc_impl(size, CallerWillInitializeMemory::No);
+    if (s_profiling)
+        perf_event(PERF_EVENT_MALLOC, size, reinterpret_cast<FlatPtr>(result.value_or(nullptr)));
```

Comment:
Not sure I'm a fan of giving a bunch of PERF_EVENT_MALLOC for nullptr when we have errors, but I'm not a fan of ignoring those cases either, since allocations still take time, even if they fail. Not sure what to do tbh

Reply:
Let's not report a `PERF_EVENT_MALLOC` if the malloc fails. The purpose of logging the event is to record that memory allocation happened, after all. :)

Same comment for all other instances of the same thing.